app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java

```diff
@@ -171,7 +171,7 @@ public void clickingOnActivityMakesCorrectIntentWhenLogged() {
             public void run() {
                 mapFragment.displayNearbyMarkers();
                 if (mapFragment.googleMap != null) {
-                    mapFragment.onMarkerClick(mapFragment.activitiesMarkers.get(0));
+                    mapFragment.onMarkerClick(mapFragment.activitiesMarkerMap.get(act));
```

Comment:
Is this change (.get(0) --> .get(act)) related to your optimisation for fetching *only* the updated activity-ies?

Reply:
At first, I implemented a list of markers in the map fragment, only in the interest of making this test work (as I can't have access to a marker directly from the map). But, with the new implementation, I had to associate each activities with their respective markers, so that I could only update the marker of the activity that changed. Doing that as a HashMap, I figured that I could directly use this map instead of a list in this test, killing two birds with one stone.

Reply:
At first, I implemented a list of markers in the map fragment, only in the interest of making this test work (as I can't have access to a marker directly from the map). But, with the new implementation, I had to associate each activities with their respective markers, so that I could only update the marker of the activity that changed. Doing that as a HashMap, I figured that I could directly use this map instead of a list in this test, killing two birds with one stone.